framework/src/problems/FEProblemBase.C

```diff
@@ -8991,7 +8994,36 @@ FEProblemBase::getNonlinearConvergenceNames() const
   if (_set_nonlinear_convergence_names)
     return _nonlinear_convergence_names;
   else
-    mooseError("The nonlinear convergence name(s) have not been set.");
+    mooseError("The nonlinear system convergence name(s) have not been set.");
+}
+
+bool
+FEProblemBase::hasLinearConvergenceObjects() const
+{
+  if (_set_linear_convergence_names)
```

Comment:
Do you need this boolean member? Can't you just return `_linear_convergence_names.size()`?

Reply:
the problem is whether we want to detect whether it has been set and is empty, or whether it has not been set and that s why it's empty.

This is what the nonlinear convergence are doing

Reply:
The empty case for nonlinear is usually temporary, before we add the default. 

I could see the linear ones getting a default at some point too